### PR TITLE
feat(react-navigation) added welcome page bottom tabs

### PR DESCRIPTION
**Pull request by Calinteodor**
New react navigation bottom tabs on Welcome page

### CODE REVIEW

**Review by saghul — CHANGES_REQUESTED** [review]
Left some comments, PTAL. One change I noticed from the current implementation is that we used to have a shadow / tint overlay when the typing box shows up, that is gone now. Can we bring it back?

File: react/features/recent-list/components/RecentList.native.js [hunk 1]
```diff
@@ -82,13 +85,15 @@ class RecentList extends AbstractRecentList<Props> {
...
 
         return (
-            <NavigateSectionList
-                disabled = { disabled }
-                onLongPress = { this._onLongPress }
-                onPress = { this._onPress }
-                renderListEmptyComponent
-                    = { this._getRenderListEmptyComponent() }
-                sections = { recentList } />
+            <LocalVideoTrackUnderlay style = { styles.recentList }>
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/recent-list/components/RecentList.native.js:88]: image

File: react/features/welcome/constants.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import React from 'react';
+
+import { Icon, IconEventNote, IconRestore } from '../base/icons';
+import BaseTheme from '../base/ui/components/BaseTheme';
+
+export const INACTIVE_TAB_COLOR = 'rgba(0, 0, 0, 0.4)';
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/welcome/constants.js]: Updated.

File: react/features/welcome/constants.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        backgroundColor: BaseTheme.palette.screen01Header
+    }
+};
+
+export const recentListTabBarOptions = {
+    // $FlowExpectedError
+    tabBarIcon: ({ focused }) => (
+        <Icon
+            color = { focused ? BaseTheme.palette.icon01 : INACTIVE_TAB_COLOR }
+            size = { 24 }
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/welcome/constants.js]: Created TabIcon.

File: react/features/calendar-sync/components/CalendarList.native.js [hunk 4]
```diff
@@ -75,10 +77,12 @@ class CalendarList extends AbstractPage<Props> {
 
         return (
             CalendarListContent
-                ? <CalendarListContent
-                    disabled = { disabled }
-                    listEmptyComponent
-                        = { this._getRenderListEmptyComponent() } />
+                ? <LocalVideoTrackUnderlay style = { styles.calendarSync }>
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/calendar-sync/components/CalendarList.native.js]: But there is an extra one in WelcomePage.native.js

File: config.js [hunk 5]
```diff
@@ -900,55 +900,25 @@ var config = {
      If there is no url set or there are missing fields, the defaults are applied.
      The config file should be in JSON.
      None of the fields are mandatory and the response must have the shape:
-    {
```
Comment (saghul) [inline config.js:903]: This looks like some merge error

File: react/features/welcome/components/WelcomePage.native.js [hunk 6]
```diff
@@ -381,10 +380,10 @@ class WelcomePage extends AbstractWelcomePage<*> {
                                 }
                             </View>
                         </SafeAreaView>
-                        {/* $FlowExpectedError*/}
-                        <WelcomePageLists disabled = { this.state._fieldFocused } />
+                        {/* // $FlowExpectedError*/}
```
Comment (saghul) [inline react/features/welcome/components/WelcomePage.native.js:383]: Is this necessary?

File: react/features/welcome/components/WelcomePage.native.js [hunk 7]
```diff
@@ -19,21 +19,20 @@ import { MEDIA_TYPE } from '../../base/media';
...
+import BaseTheme from '../../base/ui/components/BaseTheme.native';
 
 import {
     AbstractWelcomePage,
     _mapStateToProps as _abstractMapStateToProps,
     type Props as AbstractProps
 } from './AbstractWelcomePage';
-import LocalVideoTrackUnderlay from './LocalVideoTrackUnderlay';
 import VideoSwitch from './VideoSwitch';
-import WelcomePageLists from './WelcomePageLists';
```
Comment (saghul) [inline react/features/welcome/components/WelcomePage.native.js:36]: You haven't deleted this file.

File: react/features/welcome/components/WelcomePage.native.js [hunk 8]
```diff
@@ -346,7 +345,7 @@ class WelcomePage extends AbstractWelcomePage<*> {
         return (
             <>
                 <JitsiStatusBar />
-                <LocalVideoTrackUnderlay style = { styles.welcomePage }>
+                <View style = { styles.welcomePage }>
```
Comment (saghul) [inline react/features/welcome/components/WelcomePage.native.js:348]: Do we need this view? Maybe we can combine these 2 views?

File: react/features/recent-list/components/RecentList.native.js [hunk 9]
```diff
@@ -82,13 +84,15 @@ class RecentList extends AbstractRecentList<Props> {
...
 
         return (
-            <NavigateSectionList
-                disabled = { disabled }
-                onLongPress = { this._onLongPress }
-                onPress = { this._onPress }
-                renderListEmptyComponent
-                    = { this._getRenderListEmptyComponent() }
-                sections = { recentList } />
+            <View style = { disabled ? styles.recentListDisabled : styles.recentList }>
```
Comment (saghul) [inline react/features/recent-list/components/RecentList.native.js:87]: Doesn't a similar thing need to be applied for the calendar tab?

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/welcome/components/WelcomePage.native.js (hunk 6 above)
Comment (Calinteodor) [reply react/features/welcome/components/WelcomePage.native.js:383]: Yes, flow was at it again.

File: react/features/welcome/components/WelcomePage.native.js (hunk 7 above)
Comment (Calinteodor) [reply react/features/welcome/components/WelcomePage.native.js:36]: Deleted.

File: react/features/welcome/components/WelcomePage.native.js (hunk 8 above)
Comment (Calinteodor) [reply react/features/welcome/components/WelcomePage.native.js:348]: I will try.

File: react/features/recent-list/components/RecentList.native.js (hunk 9 above)
Comment (Calinteodor) [reply react/features/recent-list/components/RecentList.native.js:87]: I applied it.

File: react/features/welcome/components/WelcomePageLists.js [hunk 10]
```diff
@@ -1,139 +0,0 @@
...
-
-/**
- * The type of the React {@code Component} props of {@link WelcomePageLists}.
- */
-type Props = {
-
-    /**
-     * Whether the calendar functionality is enabled or not.
-     */
-    _calendarEnabled: boolean,
```
Comment (saghul) [inline react/features/welcome/components/WelcomePageLists.js:21]: Does the new implementation consider this?

Comment (Calinteodor) [reply react/features/welcome/components/WelcomePageLists.js:21]: Forget about this. Right now it does. Thanks!

File: react/features/welcome/components/WelcomePage.native.js (hunk 8 above)
Comment (Calinteodor) [reply react/features/welcome/components/WelcomePage.native.js:348]: Done.

File: react/features/welcome/reducer.js [hunk 11]
```diff
@@ -19,18 +14,3 @@ const STORE_NAME = 'features/welcome';
 PersistenceRegistry.register(STORE_NAME, {
     defaultPage: true
```
Comment (saghul) [inline react/features/welcome/reducer.js:15]: You can remove all of this, can't you?

Comment (Calinteodor) [reply react/features/welcome/reducer.js]: To the garbage.

**Review by saghul — APPROVED** [review]

**Pull request merged**